README.md

```diff
@@ -1,17 +1,103 @@
 Nitrokey Pro firmware [![Build Status](https://travis-ci.org/Nitrokey/nitrokey-pro-firmware.svg?branch=master)](https://travis-ci.org/Nitrokey/nitrokey-pro-firmware)  [![Code Health](https://landscape.io/github/Nitrokey/nitrokey-pro-firmware/master/landscape.svg?style=flat)](https://landscape.io/github/Nitrokey/nitrokey-pro-firmware/master) [![Coverity Scan Build](https://scan.coverity.com/projects/4745/badge.svg)](https://scan.coverity.com/projects/4745)
 =====================
 
-Building
-========
+* [Building](#building)
+* [Flashing](#flashing)
+  * [STM32flash and DFU Mode](#stm32flash-and-dfu-mode)
+    * [Requirements](#requirements)
+    * [Actual Flashing](#actual-flashing)
+  * [Versaloon](#versaloon)
+
+# Building
+
 make \[VID=0x20a0\] \[PID=0x4108\] firmware
 
 Parameters:
 * VID: Define Vendor ID
 * PID: Define Product ID
 
-Flashing
-=======================
-#### Versaloon
+# Flashing
+
+The microprocessor can be flashed in two different ways:
+* DFU is a simple protocol via serial port which allows programming but no debugging. On the Nitrokey hardware we expose the appropriate pins over the USB connector but it's not USB! Details are described in the next chapter.
+* SWD is a STM-specific protocol and similar to JTAG allowing programming and debugging. Working adapters are Versaloon or any of the ST-Link V2 (clones). Under Linux you could give a patched OpenOCD a try but in the past it has been very troublesome. This approach requires to solder wires to the contact pads or to use an adapter with pogo pins and some kind of mounting (recommended).
```

Comment:
Latest OpenOCD (v0.10) works out-of-the-box, without additional patching. I have used this method for a week month ago, on Ubuntu 18.04, and had no issues with flashing.

Reply:
I never tested the second approach, therefore I copy pasted from the previous versions...

What concrete text do you propose?